### ISSUE TITLE
Does the following code in losses.py contain a bug?

### ISSUE THREAD

**Issue by auroua** [ElementAI/LCFCN#20]
The following code from `losses.py` file from line 11 to 19.
```
    model.train()
    N =  batch["images"].size(0)
    assert N == 1

    blob_dict = get_blob_dict(model, batch)
    # put variables in cuda
    images = batch["images"].cuda()
    points = batch["points"].cuda()
    counts = batch["counts"].cuda()
```
The `get_blob_dict` set the model to `eval()`, and afterwards the model is in `eval`. The model is in `eval` during the code to calculate `loss` and  backward `loss` .
Does this correct?

**Comment by IssamLaradji** [comment]
It is not intentional, but it will not change the results. This is because I set the resnet backbone to `.eval` mode in the forward pass anyway.

**Comment by auroua** [comment]
The code is trained in `eval` mode. So the result does not change when do validation. I think the performance will improve if set the model to `train` during training. As the `ResNet` backbone use lots of `BatchNorm` layer.

**Comment by IssamLaradji** [comment]
True, let me know how well it works for you! though, it didn't work well in my experience :(

**Comment by auroua** [comment]
I will test it. Thanks for your replay.
